src/webgpu/api/validation/encoding/encoder_state.spec.ts

```diff
@@ -17,21 +17,28 @@ TODO:
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
-import { objectEquals } from '../../../../common/util/util.js';
+import { assert, objectEquals, unreachable } from '../../../../common/util/util.js';
 import { ValidationTest } from '../validation_test.js';
 
 class F extends ValidationTest {
+  private static attachmentTexture: GPUTexture | null = null;
+  private static attachmentTextureView: GPUTextureView | null = null;
```

Comment:
It's not guaranteed each test/case will use the same GPUDevice, so we should avoid caching like this. Is it measurably slower without the caching? Likely not, but it is maybe we could implement this with a WeakMap with GPUDevice keys.

Reply:
The intention is to avoid calling `createView` inside trying to catch validation error in beginRenderPass (breaking the rule that `expectValidationError` should only contain one GPU API call)
But as pointed out in the later comment, the first call to beginRenderPass will still include calling `createView`.
Since performance is not the concern here, rethinking it a bit, I should better create the texture and textureView in overrided `init` for this test set F